File: hurdles.js
   // console.log(n); // number of hurdles
   // console.log(k); // max jump height
   // console.log(height); // array of hurdles
    
    var largest = Math.max.apply(Math, height); 
        
    if (largest > k) {
      console.log(largest - k);
    } else {
      console.log(0);
    }

File: hurdles.rb
#!/bin/ruby

n,k = gets.strip.split(' ')
n = n.to_i
k = k.to_i
height = gets.strip
height = height.split(' ').map(&:to_i)

largest = height.max

if largest > k
    puts largest - k
else
    puts 0
end